pkg/internal/checkup/pod/pod_test.go

```diff
@@ -0,0 +1,193 @@
+package pod_test
+
+import (
+	"testing"
+
+	networkv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kiagnose/kubevirt-dpdk-checkup/pkg/internal/checkup/pod"
+)
+
+const testPodName = "my-pod"
+const testContainerName = "my-container"
+
+func TestCreateNetworkRequestAnnotationShouldSucceed(t *testing.T) {
+	type successTestCase struct {
+		description              string
+		inputNetworkRequestsMap  map[string]*networkv1.NetworkAttachmentDefinition
```

Comment:
Please consider to rename this map, in case you change the name of the production one.

Reply:
will keep this as agreed above